File: pinyougou-user-web/src/main/webapp/js/controller/UserController.js
app.controller("userController",function ($scope,userService) {

    $scope.password="";
    $scope.smsCode="";
    $scope.reg = function () {

      if($scope.password==$scope.entity.password){
          userService.register($scope.entity,$scope.smsCode).success(
              function (response) {
                  alert(response.message);
              }
          );
      } else{
          $scope.password="";
          $scope.entity.password="";
          alert("两次输入不一致!");
      }
    }

    $scope.sendCode = function () {
        if($scope.entity.phone==null){
            alert("手机号码不能为空！")
            return;
        }
        userService.sendCode($scope.entity.phone).success(
            function (response) {
                alert(response.message);
            }
        );
    }
});

File: pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbUser;
import com.pinyougou.user.service.UserService;
import entry.Result;
import org.springframework.util.DigestUtils;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.Date;

@RestController
@RequestMapping("/user")
public class UserController {

    @Reference
    private UserService userService;

    @RequestMapping("/register")
    public Result register(@RequestBody TbUser user, String smsCode){
        boolean flag = userService.checkCode(smsCode, user.getPhone());
        if(!flag){
            return new Result(false,"验证码不一致！");
        }
        try {
            userService.register(user);
            return new Result(true,"注册成功");
        }catch (Exception e){
            return new Result(false,"注册失败");
        }

    }

    @RequestMapping("/sendCode")
    public Result sendCode(String phone){
        try {
            boolean flag = userService.sendCode(phone);
            if(flag){
                return new Result(true,"发送成功");
            }else{
                return new Result(false,"发送失败");
            }

        }catch (Exception e){
            return new Result(false,"发送失败");
        }
    }
    
    
}
